### PR TITLE
feat(semantic pull request): add action-semantic-pull-request

### LINKED ISSUE #7164: help request:  Alternatives to Semantic Pull Requests
### Description

Currently we are using [Semantic Pull Requests](https://github.com/zeke/semantic-pull-requests ) to check in PR but the project is no longer maintained. we need some alternatives.

e.g.

* https://github.com/amannn/action-semantic-pull-request 

If anyone is interested in this, please submit a PR to complete this replacement

### PR DESCRIPTION
**Pull request by ccxhwmy**
### Description

Fixes #7164 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: .github/workflows/semantic.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  main:
+    name: Validate PR title
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - uses: ./.github/actions/action-semantic-pull-request
+        env:
```
Comment (spacewander) [inline .github/workflows/semantic.yml:20]: APISIX doesn't use the default configuration. Let's move the configuration in https://github.com/apache/apisix/blob/master/.github/semantic.yml to the new action.

Comment (ccxhwmy) [reply .github/workflows/semantic.yml:20]: > APISIX doesn't use the default configuration. Let's move the configuration in https://github.com/apache/apisix/blob/master/.github/semantic.yml to the new action.

Is the `allowRevertCommits: true` means accepting a format like `Revert "feat(xxx): xxx"`

Comment (spacewander) [reply .github/workflows/semantic.yml:20]: Yes

Comment (ccxhwmy) [reply .github/workflows/semantic.yml:20]: I add the custom types, and action-semantic-pull-request does not support the option `titleOnly` and `allowRevertCommits`, I tested and found that action-semantic-pull-request check title only defaultly.

**Review by spacewander — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**